File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }

        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }

        int result = 0;
        for (int num : set) {
            if (!set.contains(num - 1)) {
                int cur = num;
                int count = 1;
                while (set.contains(cur + 1)) {
                    count++;
                    cur++;
                }
                result = Math.max(count, result);
            }
        }

        return result;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> set;
        for (int num : nums) {
            set.insert(num);
        }

        int result = 0;
        for (int num : set) {
            if (set.find(num - 1) == set.end()) {
                int curCount = 1;
                int curElement = num;

                while (set.find(curElement + 1) != set.end()) {
                    curCount++;
                    curElement++;
                }

                result = max(result, curCount);
            }
        }

        return result;
    }
};